src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -571,6 +572,18 @@ void OrbitMainWindow::SaveCurrentTabLayoutAsDefaultInMemory() {
   }
 }
 
+void OrbitMainWindow::SaveMainWindowGeometry() {
+  QSettings settings;
+  settings.setValue(kMainWindowGeometrySettingKey, saveGeometry());
+  settings.setValue(kMainWindowStateSettingKey, saveState());
+}
+
+void OrbitMainWindow::RestoreMainWindowGeometry() {
+  QSettings settings;
+  restoreGeometry(settings.value(kMainWindowGeometrySettingKey).toByteArray());
+  restoreState(settings.value(kMainWindowStateSettingKey).toByteArray());
```

Comment:
Could this fail and crash the application if we're changing QT versions? Can you try to modify the value stored in the registry and see if this crashes?

If a QT exception is raised, this should be caught. If there's danger of undefined behavior, it might be worth storing the version alongside and deleting the saved values if the version has changed.

Reply:
Thanks for pointing out this!

I modified the value of qtVersion stored in the registry and there is no crash.
But I'm not sure whether this guarantees we are fine when changing to a new QT version.

Reply:
Unfortunately the documentation of restoreGeometry is not very clear. But I had a look at the code before because of that and I has its own versioning system. It stores a major and a minor version number. A change of the major version indicates a backwards compatible change. A change of the minor version number indicates a forwards and backwards compatible change (usually fields are appended to the end of the stream to allow the old code to continue to function).

That being said I believe we are good here. We also don't have to worry about exceptions since Qt doesn't use them at all. The function returns a bool which indicates success or failure.

One thing we might wanna check: What's the size of the main window when the user didn't have a geometry stored? Is it a reasonable default size? Should we maximize in that case?

Reply:
Ok, I was more afraid of unexpected exceptions, like Qt trying to cast the byte array into something reasonable and failing horribly.

I think it's ok to ignore the case of a first start, I've just tested it and it seems like a reasonable default size to me.

But another thing just came to my mind: What if the stored geometry is no longer on the screen, e.g. because the user changed displays, or had 2 monitors and disabled on of them? I've had that with other applications in the past and it's super annoying. You can bring back the application onto your screen by using the task bar context menu "move" and the arrow keys, but that's not very obvious.

Vicky, could you try setting a position off-screen and see what happens on restore? Maybe just programmatically set the main window position to something unreasonable on startup, close Orbit, remove the code again, and test what happens on restart.

Also: This might affect the E2E tests. We should probably maximize the window in our E2E tests first to make sure it's fully on screen.

Reply:
```
But another thing just came to my mind: What if the stored geometry is no longer on the screen, e.g. because the user changed displays, or had 2 monitors and disabled on of them? I've had that with other applications in the past and it's super annoying. You can bring back the application onto your screen by using the task bar context menu "move" and the arrow keys, but that's not very obvious.
```

That's the beauty of restoreGeometry. It handles all these cases and even the more obscure ones like taking care of multi screen setups and trying to restore the window at the exact same position as it was last time.

Reply:
Also just tried removing the QSettings in registry. The main window sizes seems to be what we set in the `.ui` file.

Thanks for all the suggestions!

I will create a follow-up PR for maximizing Orbit in the E2E tests.